### PR TITLE
Update spec guides to use new google docs v4 api

### PR DESCRIPTION
**Pull request by B77Mills**
- Add new spec-guide fetch route(__sec-guide?src=${sheetSrc}) that retrieves sheet data with api key set
 - Add table component logic to handle new data format.  Previously each row would have come back with `gsx$${key}.$t: value` array. Now it just returns an array of values.  I added logic to key each row off of row one column header. line 400-408 in packages/common/browser/spec-guide/table.vue 
 - Update FCP & OEM spec guide configs sheet src

### CODE REVIEW

File: packages/common/browser/spec-guide/table.vue [hunk 1]
```diff
@@ -392,10 +392,21 @@ export default {
...
+        // from unkeyed rows to keyed rows by column header
+        const unKeyed = get(json, 'values');
+        const sheetKeys = unKeyed.shift().map(v => v.toLowerCase().replace(/\s+|[,()/]/g, ''));
+        let rows = unKeyed.map((row) => {
+          const keyedRow = {};
+          sheetKeys.forEach((key, index) => {
+            keyedRow[key] = row[index];
+          });
+          return keyedRow;
+        });
```
Comment (solocommand) [inline packages/common/browser/spec-guide/table.vue:410]:
```js
const rows = unKeyed.map((row) => sheetKeys.reduce((obj, key, index) => ({ [key]: row[index] }), {}))
```

File: packages/common/spec-guide/index.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+module.exports = (app) => {
+  app.use('/__spec-guide', asyncRoute(async (req, res) => {
+    const { src } = req.query;
+    const url = `${src}?key=${GOOGLE_DOC_API_KEY}`;
+    try {
+      const response = await fetch(url);
+      const json = await response.json();
+      res.send(json);
+    } catch (error) {
+      res.send(error);
```
Comment (solocommand) [inline packages/common/spec-guide/index.js:14]: Send a 500 statusCode in this case

**Review by solocommand — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 3]
```diff
@@ -23,6 +23,7 @@ x-env-defaults: &env
   # @todo the recaptcha values should be removed (?) once contact-us is moved to core.
   RECAPTCHA_SITE_KEY: ${RECAPTCHA_SITE_KEY-(unset)}
   RECAPTCHA_SECRET_KEY: ${RECAPTCHA_SECRET_KEY-(unset)}
+  GOOGLE_DOC_API_KEY: ${GOOGLE_DOC_API_KEY-(unset)}
```
Comment (solocommand) [inline docker-compose.yml:26]: Can you name this `SPEC_GUIDE_DOCS_API_KEY` instead

**Review by solocommand — CHANGES_REQUESTED** [review]

**Comment by B77Mills** [comment]
[Spec guide errors](https://trello.com/c/it2OQYEW/79-spec-guide-errors)

File: packages/common/spec-guide/index.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+const { asyncRoute } = require('@parameter1/base-cms-utils');
+const fetch = require('node-fetch');
+
+const { SPEC_GUIDE_DOCS_API_KEY } = process.env;
```
Comment (solocommand) [inline packages/common/spec-guide/index.js:4]: This should pull from envalid, not the process env directly.
```suggestion
const fetch = require('node-fetch');
const { SPEC_GUIDE_DOCS_API_KEY } = require('../env');

```

**Review by solocommand — CHANGES_REQUESTED** [review]

**Review by solocommand — APPROVED** [review]

**Pull request merged**